pkg/broker/broker.go

```diff
@@ -1024,29 +1035,16 @@ func (a AnsibleBroker) Unbind(
 		log.Warning("Broker configured to *NOT* launch and run APB unbind")
 	}
 
-	//TODO should these not be handled in the subscriber if the subscriber is handling state
 	if bindExtCreds != nil {
-		err = a.dao.DeleteExtractedCredentials(bindInstance.ID.String())
-		if err != nil {
-			return nil, false, err
-		}
-	}
-
-	err = a.dao.DeleteBindInstance(bindInstance.ID.String())
-	if err != nil {
-		return nil, false, err
+		err = cleanupUnbind(&bindInstance, &serviceInstance, a.dao, true)
+	} else {
+		err = cleanupUnbind(&bindInstance, &serviceInstance, a.dao, false)
```

Comment:
I think that we should remove the if else and just call `cleanupUnbind` and pass in the bind creds to `cleanupUnbind` and do the nil check in that method. I think that might be more clear what that is doing.

Reply:
Yeah sounds good to me.